otter/integration/tests/test_convergence.py

```diff
@@ -322,7 +410,9 @@ def _choose_random_servers(self, (code, response), n):
         return random.sample(ids, n)
 
     def _delete_those_servers(self, ids, rcs):
-        """Delete each of the servers selected."""
+        """
+        Delete each of the servers selected, and save a list of the
+        ids of the deleted servers."""
```

Comment:
I had a suggestion actually for a function that handles some of this (and also helps prevent us from having to store stuff permanently):

something like ( pseudocode ):

````
oob_delete_then(self, rcs, num, callback, clb=[LB1, LB2]):
    state = yield self.get_scaling_group_state()
    ids = self._choose-random_servers(5)
    if clb:
        ips = yield (get ips)
    yield self.delete_those_servers(ids, rcs)
    expected_num = yield callback(rcs)
    result = yield defer.gatherResults(
        [self.wait_for_state(rcs, timeout, period, [HasLength(expected_num), Not(ContainsAny(...ids))])] +
        [lb.wait_for_nodes(rcs, timeout, period, [...ips]) for lb in clb])

    # unify results/failures
```

Since a lot of the workflow seems to be like - get state, pick N servers, kill them, do something else, verify that the servers have been deleted and cleaned up...